Update Dapr extension for AKS document to add details for managing Dapr versioning (#116125)
Azure/azure-content: articles/aks/dapr.md

> [!NOTE]
> If you're using Dapr OSS on your AKS cluster and would like to install the Dapr extension for AKS, read more about [how to successfully migrate to the Dapr extension][dapr-migration]. 

Create the Dapr extension, which installs Dapr on your AKS or Arc-enabled Kubernetes cluster. 

For example, install the latest version of Dapr via the Dapr extension on your AKS cluster:
```azurecli
az k8s-extension create --cluster-type managedClusters \
--cluster-name myAKSCluster \
--resource-group myResourceGroup \
--name dapr \
--extension-type Microsoft.Dapr \
--auto-upgrade-minor-version false
```

You have the option of allowing Dapr to auto-update its minor version by specifying the `--auto-upgrade-minor-version` parameter and setting the value to `true`. 

> Note, the versioning of Dapr is in the format `MAJOR.MINOR.PATCH`, which means `1.11.0` to `1.12.0` is a minor version upgrade.

Reviewer: ```suggestion
[Dapr versioning is in `MAJOR.MINOR.PATCH` format](https://docs.dapr.io/operations/support/support-versioning/#versioning), which means `1.11.0` to `1.12.0` is a _minor_ version upgrade.
```

> [!WARNING]

> > For production environments, we do not recommend enabling auto-upgrade-minor-version. This option is suitable for dev or test environments only.

Reviewer: ```suggestion
> For production environments, enabling auto-upgrade-minor-version is not recommended. This option is suitable for dev or test environments only.
```

--cluster-name myAKSCluster \
--resource-group myResourceGroup \
--name dapr \
--extension-type Microsoft.Dapr \
--auto-upgrade-minor-version false
```

### Targeting a specific Dapr version

> [!NOTE]
> Dapr is supported with a rolling window, including only the current and previous versions. It is your operational responsibility to remain up to date with these supported versions. If you have an older version of Dapr, you may have to do intermediate upgrades to get to a supported version.

The same command-line argument is used for installing a specific version of Dapr or rolling back to a previous version. Set `--auto-upgrade-minor-version` to `false` and `--version` to the version of Dapr you wish to install. If the `version` parameter is omitted, the extension installs the latest version of Dapr. For example, to use Dapr X.X.X:

```azurecli
az k8s-extension create --cluster-type managedClusters \
--cluster-name myAKSCluster \
--resource-group myResourceGroup \
--name dapr \
--extension-type Microsoft.Dapr \
--auto-upgrade-minor-version false \
--version X.X.X
```

### Configuring automatic updates to Dapr control plane

> [!WARNING]

> > For production environments, enabling auto-upgrade-minor-version is not recommended. This option is suitable for dev or test environments only.

Reviewer: ```suggestion
> You should only enable automatic updates to the Dapr control plan in dev or test environments. Auto-upgrade is not suitable for production environments.
```

> > You should only enable automatic updates to the Dapr control plan in dev or test environments. Auto-upgrade is not suitable for production environments.

Reviewer 2: ```suggestion
> You should enable automatic updates to the Dapr control plan only in dev or test environments. Auto-upgrade is not suitable for production environments.
```

> If you install Dapr without specifying a version, `--auto-upgrade-minor-version` **is automatically enabled**. This means that Dapr control plane will automatically update its minor version on new releases.

Reviewer: ```suggestion
If you install Dapr without specifying a version, `--auto-upgrade-minor-version` **is automatically enabled**, configuring the Dapr control plane to automatically update its minor version on new releases.
```

> If you install Dapr without specifying a version, `--auto-upgrade-minor-version` **is automatically enabled**, configuring the Dapr control plane to automatically update its minor version on new releases.
> You can disable auto-update by specifying the `--auto-upgrade-minor-version` parameter and setting the value to `false`. 
> [Dapr versioning is in `MAJOR.MINOR.PATCH` format](https://docs.dapr.io/operations/support/support-versioning/#versioning), which means `1.11.0` to `1.12.0` is a _minor_ version upgrade.

Reviewer 2: ```suggestion
If you install Dapr without specifying a version, `--auto-upgrade-minor-version` *is automatically enabled*, configuring the Dapr control plane to automatically update its minor version on new releases.
You can disable auto-update by specifying the `--auto-upgrade-minor-version` parameter and setting the value to `false`. 
[Dapr versioning is in `MAJOR.MINOR.PATCH` format](https://docs.dapr.io/operations/support/support-versioning/#versioning), which means `1.11.0` to `1.12.0` is a _minor_ version upgrade.
```
